### ISSUE TITLE
Finding an optimal directory structure and other package basics

### ISSUE THREAD

**Issue by spencerahill** [spencerkclark/aospy-db#4]
Continuing the discussion from https://github.com/spencerahill/aospy/issues/3 re: getting this off to the right start, in terms of fundamental: directory structure, etc.

First, some resources on starting a new project: 
- Jeff Knupp's ["Open sourcing a Python project the right way"](https://www.jeffknupp.com/blog/2013/08/16/open-sourcing-a-python-project-the-right-way/) blog post.
- This Python Guide ["Structuring Your Project"](http://docs.python-guide.org/en/latest/writing/structure/) blog post.
  These are worth reading through, even if you don't implement every single one of their suggestions.  They don't touch on relative imports much, but those are also good to use where possible within a package (to avoid namespace collisions).

Maybe a good way forward is for you to look those over when you have time and then we can talk about specific points.  Unfortunately I don't have as much time as I'd like to commit to this, nor will I for the next several months (thesis defense August 2).  But I will still contribute as much as possible.

**Comment by spencerahill** [comment]
P.S. not trying to be a nitpick with the above comment, I just have found that following the prevailing guidelines for these basic package ingredients is really important in the long run :)

**Comment by spencerkclark** [comment]
Many thanks, those are great resources.  When I get a chance I will try and improve the current setup.  I'll let you now if I have any questions.

> Unfortunately I don't have as much time as I'd like to commit to this, nor will I for the next several months (thesis defense August 2). But I will still contribute as much as possible.

No worries at all.  I'm not sure how much I will be able to get done in the midst of preparing for generals either :)

**Comment by spencerahill** [comment]
Ya good luck!  It's good then to have this current sprint where you are actively developing and we are hashing out some of the design issues...will leave us in a much better state once we get to the other side of our respective exams.

One thing that those two posts don't get into much is relative imports (e.g. `from .foo import bar`), which are standard and useful within a package for avoiding namespace conflicts.

**Comment by spencerkclark** [comment]
I updated the package structure (and Github repo) this morning to be more consistent with Jeff Knupp's post (and aospy).  I included a test directory and some test objects (and played around with the unittest framework).  I feel pretty comfortable with how that environment works now, but starting to think about what kinds of tests to do only increases the incentive to modularize the code (as in #5).  I think I will wait until I do that to beef up the testing environment.

Let me know if you have any comments on this new directory structure.  I maintained the ipython notebooks in a directory called `interactive_testing`. I think it's still useful to have an environment to play in interactively (rather than just test mode).  And while I could move that outside the package (this is sort of the `aospy_user` equivalent), this isn't really meant to be a production version of `aospy`, so I'm not so worried about including some extra code as long as it's cordoned off in a directory.
